src/test/scala/org/broadinstitute/dsde/workbench/sam/TestSupport.scala

```diff
@@ -51,9 +53,22 @@ trait PropertyBasedTesting extends PropertyChecks with Configuration with Matche
   implicit override val generatorDrivenConfig: PropertyCheckConfiguration = PropertyCheckConfiguration(minSuccessful = 3)
 }
 
-object TestSupport extends TestSupport{
+object TestSupport extends TestSupport {
   private val executor = Executors.newCachedThreadPool()
   val blockingEc = ExecutionContext.fromExecutor(executor)
+  def testCache: Cache[WorkbenchSubject, Set[String]] = {
```

Comment:
Is there some way we can test the lookup-and-evict logic?  For example:

- Assert User A is not a member of Group B
- Add User A to Group B
- Assert User A is a member of Group B
- Remove User A from Group B
- Assert User A still shows as member of Group B
- Wait for cache expiry (set it to 5 seconds or something low/reasonable)
- Assert User A is not a member of Group B

Reply:
for the moment I am content that the integration tests fail if this is not right. It is a bit difficult to test because if the cache expiry is too long the tests are slow but if it is too short the test is flakey.